src/codeu/chat/client/commandline/Chat.java

```diff
@@ -457,4 +530,371 @@ public void invoke(List<String> args) {
     // so that it can be used.
     return panel;
   }
+  
+  private Panel createInterestPanel(final UserContext user) {
+
+    final Panel panel = new Panel();
+    
+    InterestSet interests = context.getInterestSet(user.user.id);
+    int updates = 0;
+    
+    ArrayList<Bookmark> display = new ArrayList<Bookmark>();
+    ArrayList<Time> unsorted = new ArrayList<Time>();
+    
+    for (User u : interests.users){
+      HashSet<ConversationContext> newSet = conversationsOfUser(u,user);
+      
+      for(ConversationContext c : newSet){
+        boolean isCopy = false;
+        for(Bookmark b : interests.bookmarks){
+          if(!isCopy){
+            if(b.conversation.title.equals(c.conversation.title)){
+          	  isCopy = true;
+            }
+          }
+        }
+        
+        if(!isCopy){
+          unsorted.add(c.conversation.creation);
+          display.add(new Bookmark(c));
+          interests.bookmarks.add(new Bookmark(c));
+        }
+          
+        }
+       }
+       
+    if (unsorted.size() > 0){
+	  Mergesort merge = new Mergesort();
+      display = merge.sort(unsorted, display);
+    }
+       
+       
+    for(Bookmark b : display){
+    	ConversationContext c = findConversation(b.conversation.title, user);
+        System.out.println("--- new conversation "+c.conversation.title+" from "+findUsername(c.conversation.owner)+" ---");
+          if(c.firstMessage() != null){
+            for (MessageContext message = c.firstMessage();
+                           message != null;
+                           message = message.next()) {
+              System.out.println();
+              System.out.format("USER : %s\n", findUsername(message.message.author));
+              System.out.format("SENT : %s\n", message.message.creation);
+              System.out.println();
+              System.out.println(message.message.content);
+              System.out.println();
+              updates++;
+            }
+            System.out.println("---  end of conversation  ---\n");
+          } else {
+            System.out.println("---  go start that conversation!  ---\n");
+          }
+        } 
+    
+    ArrayList<Bookmark> mainDisplay = new ArrayList<Bookmark>();
+    unsorted = new ArrayList<Time>();
+	System.out.println("updates:");
+		
+    for (Bookmark b : interests.bookmarks){
```

Comment:
It would make your code cleaner to have some sense of why you're doing all this logic in here. It looks to me that it's all to make sure the user isn't shown the same updates more than once. I'd recommend either a) adding a brief comment saying that, or b) moving this code into a separate method with a descriptive name.